src/sentry/static/sentry/app/views/organizationGroupDetails/groupTagValues.tsx

```diff
@@ -1,119 +1,94 @@
+  username?: string;
+  identifier?: string;
+  ipAddress?: string;
+} & AvatarUser;
+
+type Props = {
+  group: Group;
+  location: {
+    query: object;
   };
+} & RouteComponentProps<RouteParams, {}>;
 
-  state = {
-    tagKey: null,
-    tagValueList: null,
-    loading: true,
-    error: false,
-    pageLinks: '',
+type State = {
+  tagKey: {
+    key: string;
+    name: string;
+    totalValues: number;
   };
-
-  componentDidMount() {
-    this.fetchData();
-  }
-
-  componentDidUpdate(prevProps) {
-    const queryHasChanged = !isEqual(prevProps.location.query, this.props.location.query);
-    if (queryHasChanged || prevProps.params.tagKey !== this.props.params.tagKey) {
-      this.fetchData();
-    }
-  }
-
-  fetchData = async () => {
-    const {
-      params,
-      location: {query},
-    } = this.props;
-    this.setState({
-      loading: true,
-      error: false,
-    });
-
-    const promises = [
-      this.props.api.requestPromise(`/issues/${params.groupId}/tags/${params.tagKey}/`, {
-        query,
-      }),
-      this.props.api.requestPromise(
-        `/issues/${params.groupId}/tags/${params.tagKey}/values/`,
-        {
-          query,
-          includeAllArgs: true,
-        }
-      ),
+  tagValueList: TagValue[];
+  tagValueListPageLinks: string;
+};
+
+class GroupTagValues extends AsyncComponent<
+  Props & AsyncComponent['props'],
+  State & AsyncComponent['state']
+> {
+  getEndpoints(): [string, string][] {
+    const {groupId, tagKey} = this.props.params;
+    return [
+      ['tagKey', `/issues/${groupId}/tags/${tagKey}/`],
+      ['tagValueList', `/issues/${groupId}/tags/${tagKey}/values/`],
     ];
+  }
 
-    try {
-      const [tagKey, tagValueResponse] = await Promise.all(promises);
-      const [tagValueList, , jqXHR] = tagValueResponse;
-
-      this.setState({
-        tagKey,
-        tagValueList,
-        loading: false,
-        pageLinks: jqXHR.getResponseHeader('Link'),
-      });
-    } catch (rejections) {
-      // eslint-disable-next-line no-console
-      console.error(rejections);
-      this.setState({
-        error: true,
-        loading: false,
-      });
-    }
-  };
-
-  getUserDisplayName(item) {
+  getUserDisplayName(item): string {
     return item.email || item.username || item.identifier || item.ipAddress || item.value;
   }
 
   render() {
-    if (this.state.loading) {
-      return <LoadingIndicator />;
-    } else if (this.state.error) {
-      return <LoadingError onRetry={this.fetchData} />;
-    }
-
     const {
       group,
       params: {orgId},
     } = this.props;
-    const tagKey = this.state.tagKey;
-
-    const sortedTagValueList = sortBy(
-      this.state.tagValueList,
+    const {tagKey, tagValueList, tagValueListPageLinks, loading, error} = this.state;
+    if (loading) {
+      return <LoadingIndicator />;
+    } else if (error) {
+      return <LoadingError onRetry={this.fetchData} />;
+    }
+    const sortedTagValueList: TagValue[] = sortBy(
```

Comment:
What type does `sortedTagValueList` become here? Curious if `sortBy` correctly understands the types

Reply:
Just from VSCode tooltips it looks like it passes the type along properly since the next method `.reverse()` is passed in a list of TagValues:

![image](https://user-images.githubusercontent.com/35509934/74381418-3f47d600-4da0-11ea-997f-d0454de2f1ab.png)